### ISSUE TITLE
Smart Searching in Column search not working

### ISSUE THREAD

**Issue by diesl** [jbox-web/ajax-datatables-rails#270]
DataTables' _Smart Searching Feature_ is enabled in the [Global Search](https://datatables.net/reference/api/search()) but it seems not to be enabled in the [Column Search](https://datatables.net/reference/api/column().search()). 

In DataTables' Non Server-Side Processing mode, Smart Search is always enabled by default.

The difference in Column and Global Search is confusing for users, because they expect the same behaviour for both searches. Unfortunately I also can not find an option to enable Smart Search manually for Columns.

Can you give any hints?

**Comment by diesl** [comment]
Why did you close this issue as invalid?

I think this issue is legitimate, because DataTables uses SmartSearch in Columns as default, while ajax-datatables-rails does not. In my opinion, they should behave the same.

**Comment by n-rodriguez** [comment]
> I think this issue is legitimate, because DataTables uses SmartSearch in Columns as default, while ajax-datatables-rails does not. In my opinion, they should behave the same.

I don't know how it's related to this gem.

**Comment by diesl** [comment]
> I don't know how it's related to this gem.

Because the Gem does create the SQL query that fetches the data according to the search terms.

So basically the smart search does split the input using the `global_search_delimiter` and then performs a search with each of the splitted terms. The only difference is that a smart column search is just for the particular column, whereas global smart search has to performed for every column marked as _searchable_, obviously. The global search works as expected, just the column search takes the whole string as search term.

To sum it up:

```sql 
# Current implementation:
... WHERE column LIKE '%term1 term2%'

# Smart search: 
... WHERE column LIKE '%term1%' AND column LIKE '%term2%'
```

I hope I could make things more clear.

**Comment by n-rodriguez** [comment]
> I hope I could make things more clear.

Yes, thank you!
